### PR TITLE
Set options on the command line and in config.rb

### PR DESCRIPTION
**Pull request by dometto**
Default file that defines the projects to be monitored is now: `projects.yml`. User can also define a `config.rb` file using the `--config` switch. An example file is included in the repository.

### CODE REVIEW

File: bin/cutting_edge [hunk 1]
```diff
@@ -1,57 +1,87 @@
...
-store = Moneta.new(:Memory)
-CuttingEdge::App.set(:store, store)
+opts = OptionParser.new do |opts|
+  opts.banner = 'CuttingEdge is a dependency monitoring application.
+  
+  Usage:
+      cutting_edge [options] [projects]
+      
+  Arguments:
+      [projects]                     Path to the YAML file which defines the projects to be monitored. If not specified, projects.yml in the current working directory is used.
```
Comment (bartkamphorst) [inline bin/cutting_edge:23]: Indentation seems off.

File: bin/cutting_edge [hunk 2]
```diff
@@ -1,57 +1,87 @@
...
+'
+  opts.separator ''
+  
+  opts.on('-h', '--host [HOST]', 'Specify the hostname or IP address to listen on. Default: \'0.0.0.0\'.') do |host|
+    options[:bind] = host
+  end
+  
+  opts.on('-p', '--port [PORT]', 'Specify the port to bind Gollum with. Default: \'4567\'.') do |port|
```
Comment (bartkamphorst) [inline bin/cutting_edge]: Gollum? 😅 CuttingEdge!

File: config.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#    outdated_patch: '#dfb317',
+#    outdated_minor: '#fe7d37',
+#    outdated_major: '#e05d44',
+#    unknown: '#9f9f9f'
+#   } # Redefine the colors used in SVG badges
+# 
+#   # The following will allow you to define projects on your own GitLab server in projects.yml
+#   # If you want to use the key 'myserver' in projects.yml, your repository class must be named MyserverRepository
+#   require './lib/cutting_edge/repo.rb'
+#   class MyserverRepository < GitlabRepository # The naming here is .
```
Comment (bartkamphorst) [inline config.rb]: "The naming here is ." ?

**Review by bartkamphorst — CHANGES_REQUESTED** [review]
Just a couple of minor comments.

File: bin/cutting_edge (hunk 2 above)
Comment (dometto) [reply bin/cutting_edge]: Oops! 😆

File: bin/cutting_edge (hunk 1 above)
Comment (dometto) [reply bin/cutting_edge:23]: Yeah it seems weird, but oddly it's the indentation that's required to get everything to align in the terminal output 😕

**Comment by dometto** [comment]
@bartkamphorst I've added the `url_for_project` method here. This required me to restructure the way in which subclasses of `Repository` functioned, and take a new approach to inheritance/defining your own gitlab server.

Basically, instead of making custom gitlab servers subclasses of the `GitlabRepository` class, they are now subclasses of the `Repository` class that use the `GitlabMixin` mixin. All gitlab servers, including GitLab.com, are thus defined dynamically using the new `define_gitlab_server` method. This is also much more user friendly for users who want to define their own gitlab instance.

The new dynamic approach made it super cheap to add support for custom [Gitea](https://try.gitea.io) servers, too. :)

**Review by bartkamphorst — APPROVED** [review]
Looks good.

**Pull request merged**